pkg/proxy/util/nfacct/nfacct_linux_test.go

```
@@ -478,7 +479,12 @@ func TestRunner_List(t *testing.T) {
 	rnr, err := newInternal(hndlr)
 	assert.NoError(t, err)
 
-	counters, err := rnr.List()
+	var counters []*Counter
+
+	util.RetryOnIntr(func() error {
```

We don't need to handle this here you have already handled this in `nfacct_linux.go`. 
We can actually mock `unix.EINTR` in the `TestRunner_List` and `TestRunner_Get`.

For this test we can populate fake handler with mock data for two requests, first one should error out with `unix.EINTR` and second one should succeed and return the mock response. 

Since we are retrying on `unix.EINTR` we should not expect any error from `rnr.List()` here, we can also assert on the number of calls made.

---

> We don't need to handle this here you have already handled this in `nfacct_linux.go`.

`nfacct_linux.go` is just bubbling the error up, hence the retry being handled here. Should I move the retry over to `nfacct_linux.go` ?

> For this test we can populate fake handler with mock data for two requests, first one should error out with unix.EINTR and second one should succeed and return the mock response.

I'll give that a go, thanks

---

Yes, the retry logic needs to go to `nfacct_linux.go`. I assumed it is already there, sorry for the confusion.

---

> We can actually mock `unix.EINTR` in the `TestRunner_List` and `TestRunner_Get`.

Just want to double check this.
From my understanding, List() makes sense to mock, see: https://github.com/kubernetes/kubernetes/blob/1ee2d3ca71f2f79c325ee92902fb92ea607b014c/pkg/proxy/util/nfacct/nfacct_linux.go#L149-L150
It is using `unix.NLM_F_DUMP`, so a partial response could be returned.

As for Get(): 
https://github.com/kubernetes/kubernetes/blob/1ee2d3ca71f2f79c325ee92902fb92ea607b014c/pkg/proxy/util/nfacct/nfacct_linux.go#L130-L131

It's not using a flag that may result in a partial result, so I assume there's no need to do anything there.

Does my logic make sense here?